src/Illuminate/Collections/Traits/EnumeratesValues.php

```diff
@@ -384,8 +384,11 @@ public function mapToGroups(callable $callback)
     /**
      * Map a collection and flatten the result by a single level.
      *
-     * @param  callable(TValue, TKey): mixed  $callback
-     * @return static<int, mixed>
+     * @template TFlatMapKey of array-key
+     * @template TFlatMapValue
+     *
+     * @param  callable(TValue, TKey): array<TFlatMapKey, TFlatMapValue>  $callback
```

Comment:
I think that the following would be nice because `Arr::collapse()` allow also `Collection`.

https://github.com/laravel/framework/blob/2794f8139c46036db22054c0d3459badadbf3f9d/src/Illuminate/Collections/Arr.php#L42-L62

```suggestion
     * @param  callable(TValue, TKey): \Illuminate\Support\Collection<TFlatMapKey, TFlatMapValue>|array<TFlatMapKey, TFlatMapValue>  $callback
```

Reply:
good spot, thanks